src/js/angular/ttyg/services/chat-message.mapper.js

```diff
@@ -12,6 +13,62 @@ export const chatMessageModelListMapper = (data) => {
     return data.map((chatMessage) => chatMessageModelMapper(chatMessage));
 };
 
+/**
+ * Maps a list of chat messages into a list of `ChatItemModel` objects, each containing a question (from the user) and its corresponding answer (from the agent).
+ * The backend returns messages as individual strings, and this function converts the messages into objects that hold the question and its answer.
+ *
+ * @param {Array<Object>} data - An array of message objects where each message contains details like the role (user or assistant) and message content.
+ * @return {Array<ChatItemModel>} A list of `ChatItemModel` instances, each containing a question and its corresponding answer.
+ *
+ * @example
+ * // Example message data from the backend
+ * const data = [
+ *   { role: 'user', conversationId: '123', agentId: 'agent-1', message: 'What is the weather?', timestamp: 1628876583 },
+ *   { role: 'assistant', conversationId: '123', agentId: 'agent-1', message: 'The weather is sunny.', timestamp: 1628876590 }
+ * ];
+ *
+ * // Example of returned array
+ * const items = [
+ * {
+ *     chatId: 123,
+ *     agentId: 'agent-1',
+ *     question: {
+ *          role: 'user',
+ *          conversationId: '123',
+ *          agentId: 'agent-1',
```

Comment:
Seems like there is some redundancy here regarding this agentid. It appears on the root level and again in the question and the answer. On the other side, it looks like the actual model is OK though. If this jsdoc is auto generated by the AI, please revise it

Reply:
My mistake, I will move agentId and conversationId from the question and answer level to the root level.

Reply:
Fixed